pkg/operator/controller/gatewayapi/crds.go

```diff
@@ -69,6 +79,22 @@ func (r *reconciler) ensureGatewayAPICRDs(ctx context.Context) error {
 	return utilerrors.NewAggregate(errs)
 }
 
+// listUnmanagedGatewayAPICRDs returns a list of unmanaged Gateway API CRDs
+// which exist in the cluster. A Gateway API CRD has "gateway.networking.k8s.io"
+// or "gateway.networking.x-k8s.io" in its "spec.group" field.
+func (r *reconciler) listUnmanagedGatewayAPICRDs(ctx context.Context) ([]string, error) {
+	gatewayAPICRDs := &apiextensionsv1.CustomResourceDefinitionList{}
+	if err := r.cache.List(ctx, gatewayAPICRDs, client.MatchingFields{gatewayAPICRDIndexFieldName: unmanagedGatewayAPICRDIndexFieldValue}); err != nil {
```

Comment:
Is the order of the items that `List` returns stable?  It might be prudent to sort the list to avoid flip-flopping `status.extension` between `{"unmanagedGatewayAPICRDNames":"foo,bar"}` and `{"unmanagedGatewayAPICRDNames":"bar,foo"}`.

Reply:
> Is the order of the items that List returns stable?

I doubt. Let me add the sorting.

Reply:
The sorting is added.